oem/zt/zt.go

```diff
@@ -0,0 +1,95 @@
+//
+// SPDX-License-Identifier: BSD-3-Clause
+//
+
+package zt
+
+import (
+	"encoding/json"
+	"fmt"
+	"net/http"
+
+	"github.com/stmcginnis/gofish/common"
+)
+
+var (
+	SubmitTestEventTarget = "/redfish/v1/EventService/Actions/EventService.SendTestEvent"
+)
+
+type SubscriptionZtRequestType struct {
+	Destination string            `json:"Destination"`
+	HTTPHeaders map[string]string `json:"HttpHeaders,omitempty"`
+	Protocol    string            `json:"Protocol,omitempty"`
+	Context     string            `json:"Context,omitempty"`
+}
+
+// subscribeResponseType zt uses a unique subscription response.
+type subscribeZtResponseType struct {
+	ODataContext         string `json:"@odata.context"`
+	ODataEtag            string `json:"@odata.etag"`
+	ODataID              string `json:"@odata.id"`
+	ODataType            string `json:"@odata.type"`
+	Context              string `json:"Context"`
+	DeliveryRetryPolicy  string `json:"DeliveryRetryPolicy"`
+	Description          string `json:"Description"`
+	Destination          string `json:"Destination"`
+	EventFormatType      string `json:"EventFormatType"`
+	ID                   int    `json:"ID"`
+	Name                 string `json:"Name"`
+	Protocol             string `json:"Protocol"`
+	common.Status        `json:"Status"`
+	SubordinateResources bool   `json:"SubordinateResources"`
+	SubscriptionType     string `json:"SubscriptionType"`
+}
+
+type eventPayload struct {
+	MessageID string `json:"MessageId"`
+}
+
+func getSubscriptionURL(ztSubscribeResponse *subscribeZtResponseType) string {
+	return fmt.Sprintf("%s/%v", ztSubscribeResponse.ODataID, ztSubscribeResponse.ID)
+}
+
+func Subscribe(c common.Client, subscriptionURL, eventsReceiverURL, protocol, eventContext string) (string, error) {
+	z := &SubscriptionZtRequestType{
+		Destination: eventsReceiverURL,
+		Protocol:    protocol,
+		Context:     eventContext,
+	}
+	resp, err := c.Post(subscriptionURL, z)
+
+	if err != nil {
+		return "", fmt.Errorf("failed to POST subscribe request to redfish due to %w", err)
+	}
+	defer resp.Body.Close()
+
+	var ztSubscribeResponse subscribeZtResponseType
+	err = json.NewDecoder(resp.Body).Decode(&ztSubscribeResponse)
+	if err != nil {
+		return "", fmt.Errorf("failed to read response body from subscription request due to: %w", err)
+	}
+
+	subscriptionLink := getSubscriptionURL(&ztSubscribeResponse)
+	return subscriptionLink, nil
+}
+
+// SubmitTestEvent sends event according to msgId and returns error.
+func SubmitTestEvent(client common.Client, msgID string) error {
```

Comment:
Sorry, looking closer here and notice this isn't quite right yet either. Like the example given earlier, these calls need to be done on OEM versions of the standard objects. So you would get an `EventService` instance, then to access the OEM specific implementations for that you would need to get the right version. Something like:

```go
eventService, _ := x.EventService()
ztEventService, _ := zt.FromEventService(eventService)
ztEventService.SubmitTestEvent("my message")
```

So there still needs to be a `ZTEventService` defined, then this would be a call on an instance of that object.

https://github.com/stmcginnis/gofish/blob/main/redfish/eventservice.go#L334

Reply:
I think its ok now. please provide feedback if not